### ISSUE TITLE
Can't install on Mac M1 by poetry

### ISSUE THREAD

**Issue by markowanga** [sdv-dev/SDV#952]
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 0.16.0
* Python version: 3.9
* Operating System: macOS Monterey

### Error Description

When I want to install newest version sdv on macOS I have error

### Steps to reproduce

```
(sdv-scripts) marcinwatroba@MacBook-Pro-Marcin scripts % poetry add sdv                                
Using version ^0.16.0 for sdv

Updating dependencies
Resolving dependencies... (0.9s)

Writing lock file

Package operations: 0 installs, 5 updates, 0 removals

  • Updating llvmlite (0.39.0 -> 0.36.0): Failed

  EnvCommandError

  Command ['/Users/marcinwatroba/.conda/envs/sdv-scripts/bin/pip', 'install', '--no-deps', '-U', 'file:///Users/marcinwatroba/Library/Caches/pypoetry/artifacts/b5/e2/c5/346cbf5b80a67dd0811dfdfbf228a1f3a9cf1dd532b95efe24bd34ac57/llvmlite-0.36.0.tar.gz'] errored with the following return code 1, and output: 
  Processing /Users/marcinwatroba/Library/Caches/pypoetry/artifacts/b5/e2/c5/346cbf5b80a67dd0811dfdfbf228a1f3a9cf1dd532b95efe24bd34ac57/llvmlite-0.36.0.tar.gz
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Building wheels for collected packages: llvmlite
    Building wheel for llvmlite (setup.py): started
    Building wheel for llvmlite (setup.py): finished with status 'error'
    error: subprocess-exited-with-error
    
    × python setup.py bdist_wheel did not run successfully.
    │ exit code: 1
    ╰─> [15 lines of output]
        running bdist_wheel
        /Users/marcinwatroba/.conda/envs/sdv-scripts/bin/python /private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py
        LLVM version... 14.0.6
        
        Traceback (most recent call last):
          File "/private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py", line 220, in <module>
            main()
          File "/private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py", line 214, in main
            main_posix('osx', '.dylib')
          File "/private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py", line 172, in main_posix
            raise RuntimeError(msg)
        RuntimeError: Building llvmlite requires LLVM 10.0.x or 9.0.x, got '14.0.6'. Be sure to set LLVM_CONFIG to the right executable path.
        Read the documentation at http://llvmlite.pydata.org/ for more information about building llvmlite.
        
        error: command '/Users/marcinwatroba/.conda/envs/sdv-scripts/bin/python' failed with exit code 1
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
    ERROR: Failed building wheel for llvmlite
    Running setup.py clean for llvmlite
  Failed to build llvmlite
  Installing collected packages: llvmlite
    Attempting uninstall: llvmlite
      Found existing installation: llvmlite 0.39.0
      Uninstalling llvmlite-0.39.0:
        Successfully uninstalled llvmlite-0.39.0
    Running setup.py install for llvmlite: started
    Running setup.py install for llvmlite: finished with status 'error'
    error: subprocess-exited-with-error
    
    × Running setup.py install for llvmlite did not run successfully.
    │ exit code: 1
    ╰─> [20 lines of output]
        running install
        /Users/marcinwatroba/.conda/envs/sdv-scripts/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
          warnings.warn(
        running build
        got version from file /private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/llvmlite/_version.py {'version': '0.36.0', 'full': 'e6bb8d137d922bec8beeb01a237254778759becd'}
        running build_ext
        /Users/marcinwatroba/.conda/envs/sdv-scripts/bin/python /private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py
        LLVM version... 14.0.6
        
        Traceback (most recent call last):
          File "/private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py", line 220, in <module>
            main()
          File "/private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py", line 214, in main
            main_posix('osx', '.dylib')
          File "/private/var/folders/t8/4j9s5lbj1cbbn0xj92r0g31c0000gn/T/pip-req-build-akhnnzx5/ffi/build.py", line 172, in main_posix
            raise RuntimeError(msg)
        RuntimeError: Building llvmlite requires LLVM 10.0.x or 9.0.x, got '14.0.6'. Be sure to set LLVM_CONFIG to the right executable path.
        Read the documentation at http://llvmlite.pydata.org/ for more information about building llvmlite.
        
        error: command '/Users/marcinwatroba/.conda/envs/sdv-scripts/bin/python' failed with exit code 1
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
    Rolling back uninstall of llvmlite
    Moving to /Users/marcinwatroba/.conda/envs/sdv-scripts/lib/python3.9/site-packages/llvmlite-0.39.0.dist-info/
     from /Users/marcinwatroba/.conda/envs/sdv-scripts/lib/python3.9/site-packages/~lvmlite-0.39.0.dist-info
    Moving to /Users/marcinwatroba/.conda/envs/sdv-scripts/lib/python3.9/site-packages/llvmlite/
     from /Users/marcinwatroba/.conda/envs/sdv-scripts/lib/python3.9/site-packages/~lvmlite
  error: legacy-install-failure
  
  × Encountered error while trying to install package.
  ╰─> llvmlite
  
  note: This is an issue with the package mentioned above, not pip.
  hint: See above for output from the failure.
  

  at ~/.poetry/lib/poetry/utils/env.py:1195 in _run
      1191│                 output = subprocess.check_output(
      1192│                     cmd, stderr=subprocess.STDOUT, **kwargs
      1193│                 )
      1194│         except CalledProcessError as e:
    → 1195│             raise EnvCommandError(e, input=input_)
      1196│ 
      1197│         return decode(output)
      1198│ 
      1199│     def execute(self, bin, *args, **kwargs):


Failed to add packages, reverting the pyproject.toml file to its original content.
(sdv-scripts) marcinwatroba@MacBook-Pro-Marcin scripts % 

```

**Comment by npatki** [comment]
Hi @markowanga, nice to meet you. I haven't used `poetry` but we can help you investigate.

For debugging, could you confirm if the following are true:

1. Are you able to successfully install other data science libraries with the same setup? For example, does `poetry add numpy` successfully work?
2. Are you able successfully install previous versions of the SDV using the same setup? For eg, you can try the previous `poetry add "sdv==0.15.0"`

**Comment by markowanga** [comment]
Yes, many libs works ok -- like numpy or pandas.

I think that problem cause
```
Updating llvmlite (0.39.0 -> 0.36.0): Failed
```

I can't install `llvmlite` with lower version, is it possible to upgrade version of `llvmlite`?

**Comment by npatki** [comment]
@markowanga there aren't any dependencies from the SDV side that should prevent you from using a higher version of `llvmlite`. Here is the relevant branch of the dependency tree:

```
sdv==0.16.0 -> sdmetrics(>=0.5.1.dev0,<0.6) -> pyts(>=0.12.0,<0.13.0) -> numba(>=0.48.0) -> llvmlite(various)
```

This may be an issue with dependencies between `numba`, `llvmlite` and the M1 chip. I would try first upgrading to the latest version of `numba` (0.56.0). Once that's successful, install `sdv`.

The other good news is that future versions our libraries will no longer require `pyts` so the issue should go away on its own once we release the next sdv v0.17.0. (Likely to happen within the next few weeks.)

**Comment by npatki** [comment]
Hi @markowanga, we've just released SDV 0.17.0, which drops the conflicting dependencies.

Closing this issue as resolved. Please feel free to reply if you're continuing to experience problems and I'll reopen to investigate.
